addOns/automation/src/main/java/org/zaproxy/addon/automation/AuthenticationData.java

```diff
@@ -197,6 +207,71 @@ public AuthenticationData(Context context) {
         }
     }
 
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private static void totpMethodToUser(
+            Map<String, Object> parameters, List<ContextWrapper.UserData> users) {
+        if (users == null || users.isEmpty()) {
+            return;
+        }
+
+        var user = users.get(0);
+        Object data = parameters.get("steps");
+        if (!(data instanceof List steps)) {
+            return;
+        }
+
+        for (Iterator<?> it = steps.iterator(); it.hasNext(); ) {
+            Map<String, Object> object = (Map<String, Object>) it.next();
+            if (isTotpType(object)) {
+                user.getInternalCredentials()
+                        .setTotp(
+                                JsonMapper.builder()
+                                        .build()
```

Comment:
Use a static mapper instance?

Reply:
I can but since this will not be called that often it avoids having a permanent mapper.